ledger/participant-integration-api/src/test/suite/scala/platform/store/ConversionsSpec.scala

```diff
@@ -3,54 +3,165 @@
-      val reason = domain.RejectionReason.PartyNotKnownOnLedger("Who on earth is Alice?")
-      val converted = reason.toParticipantStateRejectionReason
-      converted.code should be(Status.Code.INVALID_ARGUMENT.value())
-      converted.message should be("Party not known on ledger: Who on earth is Alice?")
+      assertConversion(domain.RejectionReason.PartyNotKnownOnLedger("reason"))(
+        v1_expectedCode = Status.Code.INVALID_ARGUMENT.value(),
+        v1_expectedMessage = "Parties not known on ledger: reason",
+        v2_expectedCode = Status.Code.NOT_FOUND.value(),
+        // TODO error codes: remove context from end of message (see [[com.daml.error.definitions.TransactionError.rpcStatus]])
+        v2_expectedMessage = "PARTY_NOT_KNOWN_ON_LEDGER(11,0): Party not known on ledger: reason",
+      )
     }
 
     "convert an 'SubmitterCannotActViaParticipant' rejection reason" in {
-      val reason = domain.RejectionReason.SubmitterCannotActViaParticipant("Wrong box.")
-      val converted = reason.toParticipantStateRejectionReason
-      converted.code should be(Status.Code.PERMISSION_DENIED.value())
-      converted.message should be("Submitted cannot act via participant: Wrong box.")
+      assertConversion(domain.RejectionReason.SubmitterCannotActViaParticipant("Wrong box."))(
+        v1_expectedCode = Status.Code.PERMISSION_DENIED.value(),
+        v1_expectedMessage = "Submitted cannot act via participant: Wrong box.",
+        v2_expectedCode = Status.Code.PERMISSION_DENIED.value(),
+        // TODO error codes: remove context from end of message (see [[com.daml.error.definitions.TransactionError.rpcStatus]])
+        v2_expectedMessage =
+          "An error occurred. Please contact the operator and inquire about the request <no-correlation-id>",
+      )
     }
 
     "convert an 'InvalidLedgerTime' rejection reason" in {
-      val reason = domain.RejectionReason.InvalidLedgerTime("Too late.")
-      val converted = reason.toParticipantStateRejectionReason
-      converted.code should be(Status.Code.ABORTED.value())
-      converted.message should be("Invalid ledger time: Too late.")
+      assertConversion(domain.RejectionReason.InvalidLedgerTime("Too late."))(
+        v1_expectedCode = Status.Code.ABORTED.value(),
+        v1_expectedMessage = "Invalid ledger time: Too late.",
+        v2_expectedCode = Status.Code.FAILED_PRECONDITION.value(),
+        // TODO error codes: remove context from end of message (see [[com.daml.error.definitions.TransactionError.rpcStatus]])
+        v2_expectedMessage = "INVALID_LEDGER_TIME(9,0): Invalid ledger time: Too late.",
+      )
     }
   }
+
+  private def assertConversion(actualRejectionReason: RejectionReason)(
+      v1_expectedCode: Int,
+      v1_expectedMessage: String,
+      v2_expectedCode: Int,
+      v2_expectedMessage: String,
+  ): Assertion = {
+    val errorFactoriesV1 = ErrorFactories(new ErrorCodesVersionSwitcher(false))
```

Comment:
To improve readability, please spell out the names of boolean parameters so it's obvious what `false` or `true` means in this context.